src/models/Entity.ts

```diff
@@ -40,6 +41,25 @@ export class Entity {
     return NetworkGetNetworkIdFromEntity(this.handle);
   }
 
+  public get State(): StateBagInterface {
+    return cfx.Entity(this.handle).state;
+  }
+
+  public AddStateBagChangeHandler(keyFilter: string, handler: StateBagChangeHandler): number {
+    let stateBagName: string;
+    if(NetworkGetEntityIsNetworked(this.handle))
+    {
+      stateBagName = `entity:${this.NetworkId}`
+    }
+    else
+    {
+      EnsureEntityStateBag(this.handle);
```

Comment:
Reading this now. It's unnecessary as underlaying fivem code will do this. Will ternary.